components/src/preact/aggregatedData/aggregate-table.tsx

```diff
@@ -0,0 +1,34 @@
+import { type FunctionComponent } from 'preact';
+
+import { type AggregateData } from '../../query/queryAggregateData';
+import { Table } from '../components/table';
+import { formatProportion } from '../shared/table/formatProportion';
+
+type AggregateTableProps = {
+    fields: string[];
+    data: AggregateData;
+};
+
+export const AggregateTable: FunctionComponent<AggregateTableProps> = ({ data, fields }) => {
+    const getHeaders = () => {
+        return [
+            ...fields.map((field) => {
+                return {
+                    name: field,
+                    sort: true,
+                };
+            }),
+            {
+                name: 'count',
+                sort: true,
+            },
+            {
+                name: 'proportion',
+                sort: true,
+                formatter: (cell: number) => formatProportion(cell),
+            },
+        ];
+    };
```

Comment:
Does this actually need to be a function? Can't we just make it a normal variable? 

```suggestion
    const headers = [
            ...fields.map((field) => {
                return {
                    name: field,
                    sort: true,
                };
            }),
            {
                name: 'count',
                sort: true,
            },
            {
                name: 'proportion',
                sort: true,
                formatter: (cell: number) => formatProportion(cell),
            },
        ];
```